flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/AbstractInvokable.java

```diff
@@ -104,9 +105,12 @@ public AbstractInvokable(Environment environment) {
      * execution failure. It can be overwritten to respond to shut down the user code properly.
      *
      * @throws Exception thrown if any exception occurs during the execution of the user code
+     * @return a future that is completed when this {@link AbstractInvokable} is fully terminated.
+     *     Note that it may never complete if the termination fails
```

Comment:
You are right, but the future may still never complete (e.g. nothing failed but task thread stuck).
WDYT about `Note that it may never complete if the invokable is stuck`?

Reply:
Yes, that sounds like a good solution.